app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt

```diff
@@ -119,9 +129,36 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
 
     private fun handleChooseFolder(folderId: Long, folderDisplayName: String) {
         selectedFolderId = folderId
+        selectedFolderDisplayName = folderDisplayName
         unreadFolder.summary = folderDisplayName
     }
 
+    private fun loadSavedState(savedInstanceState: Bundle?) {
+        if (savedInstanceState != null) {
+            if (savedInstanceState.containsKey(STATE_SELECTED_ACCOUNT_UUID)) {
+                val accountUuid = savedInstanceState.getString(STATE_SELECTED_ACCOUNT_UUID)
+                if (accountUuid != null) {
+                    handleChooseAccount(accountUuid)
+                }
+
+                if (savedInstanceState.containsKey(STATE_SELECTED_FOLDER_ID) &&
+                    savedInstanceState.containsKey(STATE_SELECTED_FOLDER_SUMMARY)) {
+                    val folderId = savedInstanceState.getLong(STATE_SELECTED_FOLDER_ID)
+                    val folderSummary = savedInstanceState.getString(STATE_SELECTED_FOLDER_SUMMARY)
+                    if (folderSummary != null) {
+                        handleChooseFolder(folderId, folderSummary)
+                    }
+                }
+            }
+        }
```

Comment:
Checking if the key exists and have an additional null check feels like a bit much. You can write it like this:

```suggestion
        val accountUuid = savedInstanceState.getString(STATE_SELECTED_ACCOUNT_UUID)
        if (accountUuid != null) {
            handleChooseAccount(accountUuid)

            val folderId = savedInstanceState.getLongOrNull(STATE_SELECTED_FOLDER_ID)
            val folderSummary = savedInstanceState.getString(STATE_SELECTED_FOLDER_SUMMARY)
            if (folderId != null && folderSummary != null) {
                handleChooseFolder(folderId, folderSummary)
            }
        }
```

with 

```kotlin
private fun Bundle.getLongOrNull(key: String): Long? {
    return if (containsKey(key)) getLong(key) else null
}
```